.github/workflows/main.yml

```
@@ -32,11 +32,15 @@ jobs:
       - name: Check generated requirements against existing requirements
         run: diff requirements.txt generated-requirements.txt
 
+      - name: Capture pipenv version
+        run: echo "::set-output name=pipenv-version::$(pipenv --version | sed 's/,//g')"
```

First off, thanks for doing this.

I think if we're gonna bother using `sed`, why not just remove all of the output that isn't the version anyway.

```suggestion
        run: echo "::set-output name=pipenv-version::$(pipenv --version | sed 's/pipenv, version //g')"
```

Also, why is the `::set-output` needed here?

---

Also, I think once everything here is resolved, we can probably squash these two commits into one.

---

The reason I just removed the comma was that, if there's any new release of Pipenv at all and the version string changes, I want the cache to be invalidated. I would have left the comma there except that GH Actions doesn't allow them in cache hash strings. It is probably low-risk to do it your way, but I don't see any real advantages to it. As I said, my preference was to just leave the version string raw, but the comma gums that up. What do you think?

That `::set-output...::` thing is a GH Actions "workflow command". You can inject certain commands through an `echo` invocation to make certain things happen. `set-output` causes whatever comes after the "closing" `::` to appear in the `steps.<id>.outputs.<output-name>` global property, and it becomes available in subsequent steps. Here I used that to inject the version string into the cache hash value. Here's the documentation for [workflow commands](https://docs.github.com/en/actions/reference/workflow-commands-for-github-actions). Very cool stuff!

---

> Also, I think once everything here is resolved, we can probably squash these two commits into one.

Agreed. I meant to mention this in a comment but forgot. When you're happy with everything, feel free to do an interactive rebase to squash those commits as needed (or, if you're feeling very pedantic, you could cherry pick the combined commit to another PR, then rebase this on top of that--but I don't really think that's necessary).

---

> The reason I just removed the comma was that, if there's any new release of Pipenv at all and the version string changes, I want the cache to be invalidated. I would have left the comma there except that GH Actions doesn't allow them in cache hash strings. It is probably low-risk to do it your way, but I don't see any real advantages to it. As I said, my preference was to just leave the version string raw, but the comma gums that up. What do you think?

The main reason I suggested it is because currently there's still a space in the cache string. It's not a huge deal, and I guess is allowed, but that was the main reason.

> That `::set-output...::` thing is a GH Actions "workflow command". You can inject certain commands through an `echo` invocation to make certain things happen. `set-output` causes whatever comes after the "closing" `::` to appear in the `steps.<id>.outputs.<output-name>` global property, and it becomes available in subsequent steps. Here I used that to inject the version string into the cache hash value. Here's the documentation for [workflow commands](https://docs.github.com/en/actions/reference/workflow-commands-for-github-actions). Very cool stuff!

:+1:

---

> > The reason I just removed the comma was that, if there's any new release of Pipenv at all and the version string changes, I want the cache to be invalidated. I would have left the comma there except that GH Actions doesn't allow them in cache hash strings. It is probably low-risk to do it your way, but I don't see any real advantages to it. As I said, my preference was to just leave the version string raw, but the comma gums that up. What do you think?
> 
> The main reason I suggested it is because currently there's still a space in the cache string. It's not a huge deal, and I guess is allowed, but that was the main reason.
> 

Since spaces are allowed, I am not worried about them remaining in the string. I'm more worried about a future version  update *changing* the format of the version output (and thus making our postprocessing logic broken somehow), so I made the minimal change to the string to get it to work (i.e., zapping the commas).